apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/AbstractSpan.java

```diff
@@ -168,7 +192,26 @@ public void setName(@Nullable String name) {
      * @return {@code this}, for chaining
      */
     public T appendToName(String s) {
-        name.append(s);
+        return appendToName(s, PRIO_DEFAULT);
+    }
+
+    public T appendToName(String s, int priority) {
+        if (priority >= namePriority) {
+            name.append(s);
```

Comment:
```suggestion
            name.append(s);
            this.namePriority = priority;
```

Add a test that fails without that